eZ/Publish/API/Repository/BookmarkService.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository;
+
+use eZ\Publish\API\Repository\Values\Bookmark\BookmarkList;
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+/**
+ * Bookmark Service.
+ */
+interface BookmarkService
+{
+    /**
+     * Add location to bookmarks.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException When location is already bookmarked
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function createBookmark(Location $location): void;
+
+    /**
+     * Removes given location from bookmarks.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function deleteBookmark(Location $location): void;
+
+    /**
+     * List bookmarked locations.
+     *
+     * @param int $offset the start offset for paging
+     * @param int $limit the number of bookmarked locations returned
+     * @return \eZ\Publish\API\Repository\Values\Bookmark\BookmarkList
```

Comment:
CS: return should be separated from params, probably can't be done via php-cs-fixer and I'm very late on documenting these standards :sweat_smile: